packages/cli-core/src/printer/printer.ts

```diff
@@ -7,6 +8,40 @@ export type RecordValue = string | number | string[] | Date;
  * The class that pretty prints to the console.
  */
 export class Printer {
+  private static instance: Printer;
+  private static minimumLogLevel: LogLevel;
+  private static writeStream: NodeJS.WriteStream;
+
+  // Properties for ellipsis animation
+  private timer: ReturnType<typeof setTimeout>;
+  private refreshRate: number;
+  private timerSet: boolean;
+
+  /**
+   * Instantiates Printer class with default configs.
+   */
+  private constructor(
+    minimumLogLevel: LogLevel,
+    writeStream: NodeJS.WriteStream
+  ) {
+    Printer.minimumLogLevel = minimumLogLevel;
+    Printer.writeStream = writeStream;
+    this.refreshRate = 500;
+  }
+
+  /**
+   * returns static singleton.
+   */
+  static make(
```

Comment:
What I meant by my other comment is we should make all the methods on this class instance methods, not static methods. eg
```ts
class Printer {
  private constructor() {}

  info: () => {}

  debug: () => {}

  etc
}

export const printer = new Printer();
```

Then the usage in other files is
```ts
import { printer } from '@aws-amplify/cli-core`

printer.info()
```

We should also only export a Printer type and have a `DefaultPrinter` that implements that type

Reply:
With the recent offline discussion, the last part "We should also only export a Printer type and have a DefaultPrinter that implements that type" is now irrelevant. We should export the Printer class.

Reply:
I'd still define `type Printer` and use it to declare type in caller packages.
Due to https://github.com/microsoft/TypeScript/issues/53558 (especially if we want to pass printer from cli to sandbox and they'd end up reading type defs from different places in node modules)

Reply:
Perhaps we need https://www.slf4j.org/api/org/slf4j/LoggerFactory.html :D